### DISCUSSION TITLE
How I speed up the weather module.

### DISCUSSION THREAD

**Discussion by Samueru-sama** [fastfetch-cli/fastfetch#733] [Show and tell]
I really like the weather module, however I assume that because it looks for the weather info on the internet on each launch there's a 300ms delay to get that info, that's very high and causes a noticeable delay while the info is being printed if you use this module.

My solution is this simple script that makes fastfetch fetch the weather info every half hour and pipes the output into a text file: 

```
#!/bin/sh

while true; do
fastfetch -s weather --weather-location Maracaibo --pipe | sed 's/Weather //g' > $XDG_CACHE_HOME/fastfetch/weatherinfo
sleep 1800
done
```

Then on the fastfetch config the weather module is replaced by this: 

```
    {
     "key": "Weather",
     "type": "command",
     "text": "cat $XDG_CACHE_HOME/fastfetch/weatherinfo"
    },
```

And that's it, now it only takes 2ms to print that info: 

![image](https://github.com/fastfetch-cli/fastfetch/assets/36420837/2ff769eb-9412-4b29-b868-0c9ea81069ef)

That info can also be used for other applications, like zenity which what I use for my calendar on polybar: 

`zenity --calendar --text="$(cat "$XDG_CACHE_HOME/fastfetch/weatherinfo")"`

![image](https://github.com/fastfetch-cli/fastfetch/assets/36420837/5314d817-7063-4c07-a266-73a6a45705e9)

**Comment by VanilleAngeles** [comment]
Great idea.Thanks.
So you could put your loop in crontab:
0/30 6-22 * * * fastfetch -s weather --weather-location Maracaibo --pipe | sed 's/Weather //g' > $XDG_CACHE_HOME/fastfetch/weatherinfo
Best regards
